content/docs/rfcs/33/README.md

```diff
@@ -0,0 +1,122 @@
+Being a P2P network on its own, it also depends on bootstrap nodes.
+The advantage of having a separate discovery network is reducing load on the bootstrap nodes as the actual work is done by randomly discovered nodes, which in turn increases decentralization.
+
+
+### w.r.t. Ethereum Discovery v5
+
+`33/WAKU2-DISCV5` spans a discovery network isolated from the Ethereum Discovery v5 network.
+Ethereum discv5 capability discovery (topic discovery) boils down to random walks and does not offer a O(log(n)) hop bound.
+Contrary to Kademlia, discv5 only maps node-ids into the DHT hash space, and not identifiers for arbitrary data.
+So, while the Ethereum discv5 network is very efficient in finding other discv5 nodes, it is not so efficient for finding discv5 nodes that have a specific property or offer specific services, e.g. Waku.
+The rarer the requested property, the longer a random walk will take until finding an appropriate node, which leads to a needle-in-the-haystack problem.
+
+Using a dedicated Waku2 discovery network, nodes can query this discovery network for a random set of nodes
+and all (well-behaving) returned nodes can serve as bootstrap nodes for other Waku2 protocols.
+
+# Semantics
+
+`33/WAKU2-DISCV5` fully inherits the [discv5 semantics](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-theory.md).
+
+
+# Wire Format Specification
+
+`33/WAKU2-DISCV5` inherits the [discv5 wire protocol](https://github.com/ethereum/devp2p/blob/master/discv5/discv5-wire.md) except for the following differences
+
+## WAKU2-Specific `protocol-id`
+
+Ethereum discv5:
+
+<pre>
+<code>
+header        = static-header || authdata
+static-header = protocol-id || version || flag || nonce || authdata-size
+protocol-id   = <b>"discv5"</b>
+version       = 0x0001
+authdata-size = uint16    -- byte length of authdata
+flag          = uint8     -- packet type identifier
+nonce         = uint96    -- nonce of message
+</code>
+</pre>
+
+`33/WAKU2-DISCV5`:
+
+<pre>
+kcode>
+header        = static-header || authdata
+static-header = protocol-id || version || flag || nonce || authdata-size
+protocol-id   = <b>"d5waku"</b>
+version       = 0x0001
+authdata-size = uint16    -- byte length of authdata
+flag          = uint8     -- packet type identifier
+nonce         = uint96    -- nonce of message
+</code>
+</pre>
+
+
+# Suggestions for Implementations
+
+Existing discv5 implementations
+
+* can be augmented to make the `protocol-id` selectable using a compile-time flag as in [link to nim-eth/discv5 once merged].
+* can be forked followed by changing the `protocol-id` string [link to go-discv5 once upstream]
+
+
+# Security Considerations
+
+
+## Eclipse attack
+TODO
+
+## Sybil attack
+TODO
+
+## Security Implications of a Separate Discovery Network
+
+A dedicated Waku discovery network can be subject to eclipse attacks if not properly secured.
```

Comment:
Would be good to briefly explain 1) the eclipse attack and how it harms the network and 2)why separating the waku discovery network from Eth discv5 make it vulnerable to this attack.